src/rules/value-no-longhand/index.js

```diff
@@ -0,0 +1,104 @@
+import valueParser from "postcss-value-parser"
+import { vendor } from "postcss"
+import {
+  cssWordIsVariable,
+  report,
+  ruleMessages,
+  validateOptions,
+} from "../../utils"
+
+export const ruleName = "value-no-longhand"
+
+export const messages = ruleMessages(ruleName, {
+  rejected: (unexpected, expected) => `Unexpected longhand value '${unexpected}' instead of '${expected}'`,
+})
+
+const shorthandableProperties = new Set([
+  "margin",
+  "padding",
+  "border-color",
+  "border-radius",
+  "border-style",
+  "border-width",
+])
+
+export default function (actual) {
+  return (root, result) => {
+    const validOptions = validateOptions(result, ruleName, { actual })
+    if (!validOptions) { return }
+
+    root.walkDecls(decl => {
+      const { prop, value } = decl
+
+      // ignore variables, and interpolation, and not shorthandable properties, and math operations and variables
+      if (cssWordIsVariable(prop)
+        || isIgnoredCharacters(value)
+        || !shorthandableProperties.has(vendor.unprefixed(prop))
+      ) { return }
+
+      const valuesToShorthand = []
+
+      valueParser(value).walk((valueNode) => {
+        if (valueNode.type !== "word") { return }
+
+        valuesToShorthand.push(valueParser.stringify(valueNode))
+      })
+
+      if (valuesToShorthand.length <= 1
+        || valuesToShorthand.length > 4
+      ) { return }
+
+      const shortestForm = condensedShorthand(...valuesToShorthand)
+      const shortestFormString = shortestForm.filter((value) => { return value }).join(" ")
+      const valuesFormString = valuesToShorthand.join(" ")
+
+      if (shortestFormString === valuesFormString) { return }
+
+      report({
+        message: messages.rejected(value, shortestFormString),
+        node: decl,
+        result,
+        ruleName,
+      })
+    })
+
+    function isIgnoredCharacters(value) {
+      return value.indexOf("+") !== -1
```

Comment:
I'd prefer an array of ignored characters and then just using `_.includes()` to check.